### PR TITLE
Go to checkout from <EligibilityWarnings> if user has no blocking holds

### PR DESCRIPTION
**Pull request by p-jackson**
#### Changes proposed in this Pull Request

Previously upgrading to a business plan from the `<EligibilityWarnings>` screen was done by clicking a `<Banner>`. In the new layout clicking "Update and continue" will take the user to the checkout.

* Clicking the "Proceed" button from the `<EligibilityWarnings>` view goes to checkout

Small chunk broken out of the larger #38277 change.

#### Testing instructions

* Create a free simple site
* Attempt to upload a plugin or theme
* Click "Upgrade and continue", you should go to the checkout
* Create a simple site with a business plan, but don't launch
* Attempt to upload a plugin or theme or adjust the hosting settings
* Button should just say "Continue" and it won't take you to the checkout

### CODE REVIEW

**Review by adamziel — APPROVED** [review]

**Comment by p-jackson** [comment]
Added [DO NOT MERGE], my intention was to wait for the uber #38277 PR to be ready before merging everything into there.

**Pull request merged**
